artifactory/services/props.go

```diff
@@ -74,6 +87,88 @@ func (sp *PropsParams) GetProps() string {
 	return sp.Props
 }
 
+func (ps *PropsService) getEncodedParam(propsParams PropsParams, isDelete bool) (string, error) {
+	var encodedParam string
+	if !isDelete {
+		props, err := utils.ParseProperties(propsParams.GetProps())
+		if err != nil {
+			return "", err
+		}
+		encodedParam = props.ToEncodedString(true)
+	} else {
+		propList := strings.Split(propsParams.GetProps(), ",")
+		for _, prop := range propList {
+			encodedParam += url.QueryEscape(prop) + ","
+		}
+		// Remove trailing comma
+		encodedParam = strings.TrimSuffix(encodedParam, ",")
+	}
+	return encodedParam, nil
+}
+
+func (ps *PropsService) addOrDeletePropertiesForRepo(propsParams PropsParams, isDelete bool, encodedParam string) (int, error) {
+	var action func(string, string, string) (*http.Response, []byte, error)
+	if isDelete {
+		action = ps.sendDeleteRequest
+	} else {
+		action = ps.sendPutRequest
+	}
+	successCounters := make([]int, ps.GetThreads())
+	producerConsumer := parallel.NewBounedRunner(ps.GetThreads(), false)
+	errorsQueue := clientutils.NewErrorsQueue(1)
+	reader := propsParams.GetReader()
+	resultItem := new(utils.ResultItem)
+	if reader.NextRecord(resultItem) == nil {
+		repoName := resultItem.Repo
+		setPropsTask := func(threadId int) error {
+			var err error
+			logMsgPrefix := clientutils.GetLogMsgPrefix(threadId, ps.IsDryRun())
+
+			restAPI := path.Join("api", "storage", repoName)
+			setPropertiesURL, err := clientutils.BuildUrl(ps.GetArtifactoryDetails().GetUrl(), restAPI, make(map[string]string))
+			if err != nil {
+				return err
+			}
+			// Because we do set/delete props on search results that took into account the
+			// recursive flag, we do not want the action itself to be recursive.
+			setPropertiesURL += "?properties=" + encodedParam + "&recursive=0"
+			resp, body, err := action(logMsgPrefix, repoName, setPropertiesURL)
+
+			if err != nil {
+				return err
+			}
+			if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusNoContent); err != nil {
+				return err
+			}
+			successCounters[threadId]++
+			return nil
+		}
+
+		_, _ = producerConsumer.AddTaskWithError(setPropsTask, errorsQueue.AddError)
```

Comment:
it looks be redundant to ignore return values from AddTaskWithError
```suggestion
		producerConsumer.AddTaskWithError(setPropsTask, errorsQueue.AddError)
```